src/ansible_builder/_target_scripts/introspect.py

```diff
@@ -1,16 +1,20 @@
 import argparse
-import importlib.metadata
 import logging
 import os
+import re
 import sys
 import yaml
 
-import requirements
-
-
 base_collections_path = '/usr/share/ansible/collections'
 logger = logging.getLogger(__name__)
 
+# regex for a comment at the start of a line, or embedded with leading space(s)
+COMMENT_RE = re.compile(r'(?:^|\s+)#.*$')
+
+# https://peps.python.org/pep-0503/#normalized-names
+REQ_NORM_RE = re.compile(r'[-_.]+')
+REQ_NAME_RE = re.compile(r'^([^-][-\w.]+)')
```

Comment:
Using this as the deciding factor on which data we pass through untouched is not enough. For example, this `REQ_NAME_RE` regex will match `git+https://git.repo/some_pkg.git#egg=SomePackage` and return `git`. If `git` happens to be in our exclude list, we'll lose this line and any others like it.